### PR TITLE
HLRC: Add delete watch action

### PR DESCRIPTION
**Pull request by ywelsch**
Relates #29827

### CODE REVIEW

**Comment by ywelsch** [comment]
@elasticmachine retest this please

**Review by hub-cap — CHANGES_REQUESTED** [review]
One Q, one doc failure. Other than that, looks fresh and clean. Great work.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherClient.java [hunk 1]
```diff
@@ -61,4 +64,31 @@ public void putWatchAsync(PutWatchRequest request, RequestOptions options,
...
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/watcher-api-delete-watch.html">
+     * the docs</a> for more.
+     * @param request the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public DeleteWatchResponse deleteWatch(DeleteWatchRequest request, RequestOptions options) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(request, RequestConverters::xPackWatcherDeleteWatch, options,
+            DeleteWatchResponse::fromXContent, singleton(404));
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherClient.java]: `RestDeleteWatchAction` returns a 404 with a body if the deleted object is not found. It's a bit unusual, but yeah, if I didn't add this here, the `RestHighLevelClient` class would puke all over itself.

File: docs/java-rest/high-level/x-pack/watcher/delete-watch.asciidoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
+[[java-rest-high-x-pack-watcher-delete-watch]]
```
[the remark this answers is not in the document]
Comment (ywelsch) [reply docs/java-rest/high-level/x-pack/watcher/delete-watch.asciidoc]: thanks, I've pushed a8b0423 which fixes the copy-paste error (I've tested it also by running your command). I've also added the put-watch and delete-watch action to the actual page (and not only right-hand-side table of contents).

**Comment by ywelsch** [comment]
Thanks for the quick review @hub-cap. I've addressed your question and the doc failure.

**Comment by ywelsch** [comment]
@elasticmachine retest this please

**Comment by ywelsch** [comment]
@elasticmachine retest this please.
@hub-cap CI fails for a million reasons, but none have to do with the PR here

File: client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherClient.java]: lol, ok. This is indeed fun.

File: docs/java-rest/high-level/x-pack/watcher/delete-watch.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (hub-cap) [reply docs/java-rest/high-level/x-pack/watcher/delete-watch.asciidoc]: excellent

**Review by hub-cap — APPROVED** [review]
CI got 99 problems, and a Delete Watch API aint one of em. :shipit:

**Comment by ywelsch** [comment]
Thanks @hub-cap

**Pull request merged**